openedx/features/content_tagging/api.py

```diff
@@ -0,0 +1,133 @@
+    enabled=True,
+    required=False,
+    allow_multiple=False,
+    allow_free_text=False,
+    taxonomy_class: Type = ContentTaxonomy,
+) -> Taxonomy:
+    """
+    Creates, saves, and returns a new Taxonomy with the given attributes.
+
+    If `taxonomy_class` not provided, then uses ContentTaxonomy.
+
+    If `org_owners` is empty/None, then the returned taxonomy is enabled for all organizations.
+    """
+    taxonomy = oel_tagging.create_taxonomy(
+        name=name,
+        description=description,
+        enabled=enabled,
+        required=required,
+        allow_multiple=allow_multiple,
+        allow_free_text=allow_free_text,
+        taxonomy_class=taxonomy_class,
+    ).cast()
+    if org_owners:
+        set_taxonomy_org_owners(taxonomy, org_owners)
+    return taxonomy
+
+
+def set_taxonomy_org_owners(
+    taxonomy: Taxonomy,
+    org_owners: List[Organization] = None,
+):
+    """
+    Updates the list of org "owners" on the given taxonomy.
+
+    If `org_owners` is empty/None, then the taxonomy is enabled for all organizations.
+    """
+    TaxonomyOrg.get_owners(taxonomy).delete()
+    if org_owners:
+        TaxonomyOrg.objects.bulk_create(
+            [
+                TaxonomyOrg(
+                    taxonomy=taxonomy,
+                    org=org,
+                    rel_type=TaxonomyOrg.RelType.OWNER,
+                )
+                for org in org_owners
+            ]
+        )
+
+
+def get_taxonomies_for_org(
+    org_owner: Organization = None, enabled=True
+) -> Iterator[Taxonomy]:
+    """
+    Generates a list of the enabled Taxonomies owned by the given org, sorted by name.
+
+    If you want the disabled Taxonomies, pass enabled=False.
+    If you want all Taxonomies (both enabled and disabled), pass enabled=None.
+    """
+    taxonomies = oel_tagging.get_taxonomies(enabled=enabled)
+    for taxonomy in taxonomies.all():
+        org_short_name = org_owner.short_name if org_owner else None
+        if TaxonomyOrg.is_owner(taxonomy, org_short_name):
+            yield taxonomy.cast()
+
+
+def get_content_tags(
+    object_id: str, taxonomy: Taxonomy = None, valid_only=True
+) -> Iterator[ContentObjectTag]:
+    """
+    Generates a list of content tags for a given object.
+
+    Pass taxonomy to limit the returned object_tags to a specific taxonomy.
+
+    Pass valid_only=False when displaying tags to content authors, so they can see invalid tags too.
+    Invalid tags will (probably) be hidden from learners.
+    """
+    for object_tag in oel_tagging.get_object_tags(
+        object_id=object_id,
+        taxonomy=taxonomy,
+        valid_only=valid_only,
+    ):
+        yield ContentObjectTag.cast(object_tag)
+
+
+def tag_content_object(
+    taxonomy: Taxonomy,
+    tags: List,
+    object_id: Union[BlockUsageLocator, LearningContextKey],
+) -> List[ContentObjectTag]:
+    """
+    Replace the existing ContentObjectTag entries for the given taxonomy + object_id with the given list of tags.
```

Comment:
I think a slightly higher level of description would be useful in this docstring.
I would suggest including something like:

This is the main API to use when you want to add/update/delete tags from a content object (e.g. an XBlock or course). It works one "Taxonomy" at a time, i.e. one field at a time, so you can set call it with `taxonomy=Keywords, tags=["gravity", "newton"]` to replace any "Keywords" [Taxonomy] tags on the given content object with "gravity" and "newton". Doing so to change the "Keywords" Taxonomy won't affect other Taxonomy's tags (other fields) on the object, such as "Language: [en]" or "Difficulty: [hard]".

... plus the other things you already have in the docstring.

Reply:
Sure -- I moved away from the "fields" nomenclature in the code because I find it very confusing in context. But I've put your comment in place here: [76d8dcc](https://github.com/openedx/edx-platform/pull/32661/commits/76d8dcca420d45000ea9ae9326c3ab9a6de30f95)

Reply:
Thanks! I did find this confusing at first, so some wording that like will be helpful I think.